src/client.ts

```diff
@@ -1,124 +1,154 @@
   ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
+    const endpoint = this.buildEndpoint(corpName, '/users');
 
-    for (const user of users) {
+    const { data } = await this.fetch(endpoint, {
+      headers: this.generateHeaders(),
+    });
+
+    for (const user of data) {
       await iteratee(user);
     }
   }
 
+  private buildEndpoint(corp: string, path?: string) {
+    if (!corp) {
+      throw new Error(`Parameter 'corp' is required, but received ${corp}`);
+    }
+
+    if (path) {
+      return `${BASE_URI}/corps/${corp}${path}`;
+    } else {
+      return `${BASE_URI}/corps/${corp}`;
+    }
+  }
+
+  private generateHeaders() {
+    const { apiUser, apiToken } = this.config;
+
+    return {
+      'x-api-user': apiUser,
+      'x-api-token': apiToken,
+      'Content-Type': 'application/json',
+    };
+  }
+
   /**
-   * Iterates each group resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
+   * Makes GET request, returning data or throws an error.
+   * @param endpoint
+   * @param options
+   * @returns Promise
    */
-  public async iterateGroups(
-    iteratee: ResourceIteratee<AcmeGroup>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const groups: AcmeGroup[] = [
-      {
-        id: 'acme-group-1',
-        name: 'Group One',
-        users: [
-          {
-            id: 'acme-user-1',
-          },
-        ],
-      },
-    ];
+  private async fetch(
+    endpoint: string,
+    options: AxiosRequestConfig,
+  ): Promise<SigSciResponseFormat> {
+    const { status, statusText, data } = await axios.get(endpoint, {
+      ...options,
+      validateStatus: () => true,
+    });
 
-    for (const group of groups) {
-      await iteratee(group);
+    if (status === 200) {
+      return data as SigSciResponseFormat;
+    } else if (status === 401) {
+      throw new IntegrationProviderAuthenticationError({
+        endpoint,
+        status: status,
+        statusText: statusText,
```

Comment:
I bet it would be useful to expose the `data` returned from the endpoint in case of a failure. Sadly, we don't have an excellent property to pass arbitrary objects... typically, we pass an existing error. 

You could try something like this ->
```suggestion
        endpoint,
        cause: new Error(data),
        status: status,
        statusText: statusText,
```

Reply:
I dig it.